router/pkg/routerconfig/s3/client.go

```diff
@@ -30,13 +31,29 @@ type ClientOptions struct {
 }
 
 func NewClient(endpoint string, options *ClientOptions) (routerconfig.Client, error) {
-
 	client := &Client{
 		options: options,
 	}
 
+	// The providers credential chain is used to allow multiple authentication methods.
+	providers := []credentials.Provider{
+		// Static credentials allow setting the access key and secret access key directly.
+		&credentials.Static{
+			Value: credentials.Value{
+				AccessKeyID:     options.AccessKeyID,
+				SecretAccessKey: options.SecretAccessKey,
+			},
+		},
+		// IAM credentials are retrieved from the EC2 nodes assumed role.
+		&credentials.IAM{
```

Comment:
We should document this here https://github.com/wundergraph/cosmo/blob/main/router/pkg/config/config.schema.json and on our [docs](https://github.com/wundergraph/cosmo-docs/blob/main/router/storage-providers.md).

Reply:
I've opened a second PR for the docs change here https://github.com/wundergraph/cosmo-docs/pull/12

Reply:
I've added a description for the s3 provider in `config.schema.json` here https://github.com/wundergraph/cosmo/pull/1250/commits/b0da79c392b655431c45a71861ebb635219efc67